src/Printer/Console.php

```diff
@@ -4,52 +4,83 @@
 
 namespace Povils\PHPMND\Printer;
 
-use JakubOnderka\PhpConsoleColor\ConsoleColor;
 use JakubOnderka\PhpConsoleHighlighter\Highlighter;
-use Povils\PHPMND\FileReportList;
+use Povils\PHPMND\DetectionResult;
 use Povils\PHPMND\HintList;
 use Symfony\Component\Console\Output\OutputInterface;
 
 class Console implements Printer
 {
     private const DEFAULT_LINE_LENGTH = 80;
 
-    public function printData(OutputInterface $output, FileReportList $fileReportList, HintList $hintList): void
+    /**
+     * @var Highlighter
+     */
+    private $highlighter;
+
+    public function __construct(Highlighter $highlighter)
+    {
+        $this->highlighter = $highlighter;
+    }
+
+    public function printData(OutputInterface $output, HintList $hintList, array $list): void
     {
         $length = (int) (`tput cols` ?: self::DEFAULT_LINE_LENGTH);
         $separator = str_repeat('-', $length);
         $output->writeln(PHP_EOL . $separator . PHP_EOL);
 
-        $total = 0;
-        foreach ($fileReportList->getFileReports() as $fileReport) {
-            $entries = $fileReport->getEntries();
-            $total += count($entries);
-            foreach ($entries as $entry) {
+        foreach ($this->groupDetectionResultPerFile($list) as $detectionResults) {
+            foreach ($detectionResults as $detection) {
                 $output->writeln(sprintf(
-                    '%s:%d  Magic number: %s',
-                    $fileReport->getFile()->getRelativePathname(),
-                    $entry['line'],
-                    $entry['value']
+                    '%s:%d. Magic number: %s',
```

Comment:
This is a breaking change, its probably better but we should wait till 3 really, Its part of the output

Reply:
I'd say that it was more like a typo here.